test/test_array.hpp

```diff
@@ -0,0 +1,235 @@
+#include "ports-of-call/array.hpp"
+
+#include <iostream>
+
+using namespace PortsOfCall;
+
+TEST_CASE("array nominal element access", "[array]") {
+  // Declare the array
+  constexpr int N = 16;
+  array<double, N> arr;
+
+  // Fill the array
+  for (int i = 0; i < N; ++i) {
+    arr[i] = i + 1;
+  }
+
+  // Can we read from it?
+  for (int i = 0; i < N; ++i) {
+    CHECK(arr[i] == i + 1);
+  }
```

Comment:
Some of these loops should be inside device kernels (e.g., `portable_for`) to ensure it works on device.